### PR TITLE
Do not mere : fix: Allow size mismatch tag run

### PR DESCRIPTION
**Pull request by sagar-qa007**
## Description
It is open issue from snapshot library. Enable allowSizeMismatch flag will not check for exact size.

Fixes #`35762`  

## Automation

/ok-to-test tags="@tag.All"

### :mag: Cypress test results

> [!CAUTION]
> 🔴 🔴 🔴 Some tests have failed.
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/10471567383>
> Commit: 77dac859033987fe8a68dca6f80039ff3aba49df
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=10471567383&attempt=1&selectiontype=test&testsstatus=failed&specsstatus=fail" target="_blank">Cypress dashboard</a>.
> Tags: @tag.All
> Spec: 
> The following are new failures, please fix them before merging the PR: <ol>
> <li>cypress/e2e/Regression/ClientSide/Widgets/Checkbox/CheckboxGroup2_spec.js
> <li>cypress/e2e/Regression/ClientSide/Widgets/TableV2/Inline_editing_3_spec.js
> <li>cypress/e2e/Regression/ServerSide/Postgres_DataTypes/Binary_Spec.ts</ol>
> <a href="https://internal.appsmith.com/app/cypress-dashboard/identified-flaky-tests-65890b3c81d7400d08fa9ee3?branch=master" target="_blank">List of identified flaky tests</a>.
> <hr>Tue, 20 Aug 2024 13:34:04 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
	- Introduced a new property, `allowSizeMismatch`, to enhance the flexibility of snapshot comparisons, allowing for size mismatches during visual testing.
	- Added support for the `@emerson-eps/cypress-image-snapshot` package, improving image snapshot testing capabilities in Cypress.

- **Impact**
	- This update improves the success criteria for snapshot testing, ensuring better adaptability in various testing scenarios and enhancing developer experience with type checking.

### CODE REVIEW

File: app/client/tsconfig.json [hunk 1]
```diff
@@ -34,6 +34,6 @@
     "noFallthroughCasesInSwitch": true,
     "verbatimModuleSyntax": true
   },
-  "include": ["./src/**/*", "./packages"],
+  "include": ["./src/**/*", "./packages", "@emerson-eps/cypress-image-snapshot"],
```
Comment (KelvinOm) [inline app/client/tsconfig.json:37]: What is it for? It seems that all changes in tsconfigs is not necessary.

**Review by KelvinOm — APPROVED** [review]

**Review by KelvinOm — CHANGES_REQUESTED** [review]
Why are the tests not falling? They're supposed to be falling, right?

**Comment by sagar-qa007** [comment]
@KelvinOm There is an [open bug](https://github.com/jaredpalmer/cypress-image-snapshot/issues/196) related to pixel mismatches in the old version of the library, which has not been updated for a considerable time. Given that our local Docker run shows no mismatches, the test should ideally pass here as well.

**Comment by KelvinOm** [comment]
> @KelvinOm There is an [open bug](https://github.com/jaredpalmer/cypress-image-snapshot/issues/196) related to pixel mismatches in the old version of the library, which has not been updated for a considerable time. Given that our local Docker run shows no mismatches, the test should ideally pass here as well.

@sagar-qa007 But tests should fall anyway, because what we have in the release now does not match what is in the screenshots.

**Pull request closed without merging**